acceptance/tests/options/custom_facts.rb

```diff
@@ -42,4 +45,32 @@
   on(agent, facter("--custom-dir '#{custom_dir}' custom_fact")) do
     assert_equal("testvalue", stdout.chomp, "Custom fact output does not match expected output")
   end
+
+  on(agent, "rm -f '#{custom_fact}'")
+
+  step "Agent #{agent}: ensure custom facts in $FACTERLIB resolve"
+  facterlib_dir = agent.tmpdir('arbitrary_dir')
+  custom_fact = File.join(facterlib_dir, 'custom_fact.rb')
+  create_remote_file(agent, custom_fact, content)
+  on(agent, "chmod +x '#{custom_fact}'")
+
+  on(agent, facter('custom_fact', :environment => { 'FACTERLIB' => facterlib_dir })) do
+    assert_equal("testvalue", stdout.chomp, "Custom fact output does not match expected output")
+  end
+
+  on(agent, "rm -rf '#{facterlib_dir}'")
```

Comment:
suggest deleting #{custom_fact} here as well so that we know we're resolving the correct custom fact in the next step/test.